airflow/jobs/scheduler_job.py

```diff
@@ -954,7 +954,7 @@ def _schedule_dag_run(
             unfinished_task_instances = (
                 session.query(TI)
                 .filter(TI.dag_id == dag_run.dag_id)
-                .filter(TI.execution_date == dag_run.execution_date)
+                .filter(TI.run_id == dag_run.run_id)
```

Comment:
Because AIP-39. This code path does not seem to be covered by any existing case?